packages/shared/src/hooks/squads/useSquads.ts

```diff
@@ -0,0 +1,88 @@
+import { useQuery, useQueryClient, QueryKey } from '@tanstack/react-query';
+import { ClientError } from 'graphql-request';
+import { Squad } from '../../graphql/sources';
+import { getCurrentUserSquads } from '../../graphql/squads';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { generateQueryKey, RequestKey, StaleTime } from '../../lib/query';
+import { useRequestProtocol } from '../useRequestProtocol';
+
+interface UseSquadsResult {
+  squads: Squad[];
+  isLoading: boolean;
+  isFetched: boolean;
+  error: ClientError | null;
+  addSquad: (squad: Squad) => Promise<void>;
+  updateSquad: (squad: Squad) => Promise<void>;
+  deleteSquad: (squadId: string) => Promise<void>;
+}
+
+const sortByName = (squads: Squad[]): Squad[] =>
+  [...squads].sort((a, b) =>
+    a.name.toLocaleLowerCase() > b.name.toLocaleLowerCase() ? 1 : -1,
+  );
+
+const getCached = async (
+  client: ReturnType<typeof useQueryClient>,
+  queryKey: QueryKey,
+): Promise<Squad[]> => {
+  return (await client.getQueryData<Squad[]>(queryKey)) || [];
+};
+
+export const useSquads = (): UseSquadsResult => {
+  const { user } = useAuthContext();
+  const client = useQueryClient();
+  const { requestMethod } = useRequestProtocol();
+  const queryKey = generateQueryKey(RequestKey.Squads, user);
+
+  const {
+    data: squads,
+    isLoading,
+    isFetched,
+    error,
+  } = useQuery<Squad[], ClientError>(
+    queryKey,
+    getCurrentUserSquads(requestMethod),
+    {
+      enabled: !!user?.id,
+      retry: false,
```

Comment:
Why did you disable retry specifically?
We also have a wrapper for `disabledRefetch` which maybe you want?

Reply:
Bad copy paste from the `useSquad` hook. I think we can retry here.